db/statistics.go

```diff
@@ -625,11 +626,28 @@ func GetValidatorIncomeHistoryChart(validator_indices []uint64, currency string,
 	return clRewardsSeries, err
 }
 
-func GetValidatorIncomeHistory(validator_indices []uint64, lowerBoundDay uint64, upperBoundDay uint64, lastFinalizedEpoch uint64) ([]types.ValidatorIncomeHistory, error) {
+func GetValidatorIncomeHistory(validatorIndices []uint64, lowerBoundDay uint64, upperBoundDay uint64, lastFinalizedEpoch uint64) ([]types.ValidatorIncomeHistory, error) {
+	if len(validatorIndices) == 0 {
+		return []types.ValidatorIncomeHistory{}, nil
+	}
+
 	if upperBoundDay == 0 {
 		upperBoundDay = 65536
 	}
-	queryValidatorsArr := pq.Array(validator_indices)
+
+	validatorIndices = utils.SortedUniqueUint64(validatorIndices)
+	validatorIndicesStr := make([]string, len(validatorIndices))
+	for i, v := range validatorIndices {
+		validatorIndicesStr[i] = fmt.Sprintf("%d", v)
+	}
+
+	validatorIndicesPqArr := pq.Array(validatorIndices)
+
+	cacheDur := time.Second * time.Duration(utils.Config.Chain.Config.SecondsPerSlot*utils.Config.Chain.Config.SlotsPerEpoch+10) // updates every epoch, keep 10sec longer
```

Comment:
Should be const

Reply:
Making this a const is not possible since variable values are involved to calculate it. I could calculate it outside the function, but then need to make sure config has been initialized already. Also I think it is good to have this cacheDuration-calculation close to where its used.

Reply:
Okay